android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java

```diff
@@ -193,9 +198,22 @@ public void onServicesDiscovered(BluetoothGatt gatt, int status) {
     }
 
     characteristic = gatt.getService(SERVICE_UUID).getCharacteristic(CHARACTERISTIC_WEBPAGE_UUID);
+    checkWebpageGzipped();
+    
     gatt.readCharacteristic(characteristic);
   }
 
+  private void checkWebpageGzipped() {
```

Comment:
It'd be nice if these both used the same characteristic, and you just inspected the content at the point when you needed it to see whether it is gzipped.